### PR TITLE
add quic-trace support

### PR DESCRIPTION
**Pull request by marten-seemann**

### CODE REVIEW

File: internal/ackhandler/interfaces.go [hunk 1]
```diff
@@ -39,6 +39,9 @@ type SentPacketHandler interface {
 
 	GetAlarmTimeout() time.Time
 	OnAlarm() error
+
+	// report some congestion statistics. For tracing only.
+	GetCurrentState() *State
```
Comment (lucas-clemente) [inline internal/ackhandler/interfaces.go]: Maybe call this stats instead, to make it more obvious that this is informational only?

File: quictrace/interface.go [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+
+	"github.com/lucas-clemente/quic-go/internal/ackhandler"
+	"github.com/lucas-clemente/quic-go/internal/protocol"
+	"github.com/lucas-clemente/quic-go/internal/wire"
+)
+
+// A Tracer traces a QUIC connection
+type Tracer interface {
+	Trace(protocol.ConnectionID, Event)
+	EmitAll() map[string][]byte
```
Comment (lucas-clemente) [inline quictrace/interface.go]: What does this method do?

File: internal/ackhandler/interfaces.go (hunk 1 above)
Comment (marten-seemann) [reply internal/ackhandler/interfaces.go]: Done.

File: quictrace/interface.go (hunk 2 above)
Comment (marten-seemann) [reply quictrace/interface.go]: It returns all traces. I'll rename it to `GetAllTraces`, and document it better.

**Pull request merged**